packages/components/src/components/button/button.tsx

```diff
@@ -1,11 +1,41 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
-import React from 'react';
-import ButtonLoading from './button_loading.jsx';
+import React, { HTMLProps, ReactNode } from 'react';
+import ButtonLoading from './button_loading';
 import Icon from '../icon';
 import Text from '../text';
 
-const ButtonGroup = ({ children, className }) => (
+type TButtonGroupProps = HTMLProps<HTMLDivElement>;
+
+type TButtonProps = HTMLProps<HTMLButtonElement> & {
+    alternate: boolean;
+    blue: boolean;
+    classNameSpan: string;
+    green: boolean;
+    has_effect: boolean;
+    icon: ReactNode;
+    is_button_toggle: boolean;
+    is_circular: boolean;
+    is_circle: boolean;
+    is_disabled: boolean;
+    is_loading: boolean;
+    is_plus: boolean;
+    is_submit_success: boolean;
+    large: boolean;
+    medium: boolean;
+    primary: boolean;
+    primary_light: boolean;
+    renderText: (param: ReactNode) => ReactNode;
+    rounded: boolean;
+    secondary: boolean;
+    small: boolean;
+    tertiary: boolean;
+    text: string;
+    transparent: boolean;
+    type: 'button' | 'submit' | 'reset';
+    wrapperClassName: string;
+};
+
+const ButtonGroup = ({ children, className }: TButtonGroupProps) => (
```

Comment:
```suggestion
const ButtonGroup = ({ children, className }: React.PropsWithChildren<HTMLDivElement>) => (
```
@jim-deriv  TButtonGroupProps is only used once, that's why I would suggest using it here directly.
Also,  React.PropsWithChildren is really better than React.HTMLProps because `children in React.HTMLProps` have type of `ReactI18NextChild | Iterable<ReactI18NextChild>`, whereas `children in React.PropsWithChildren` have type of `HTMLCollection & React.ReactNode`🙂